### PR TITLE
Bump deps

### PR DESCRIPTION
**Pull request by ericavonb**

### CODE REVIEW

File: pkg/controller/configmapcainjector/starter/starter.go [hunk 1]
```diff
@@ -2,49 +2,43 @@ package starter
...
-	// Verify that there is at least one cert in the bundle file
-	block, _ := pem.Decode(ca)
-	if block == nil {
-		return nil, fmt.Errorf("failed to parse CA bundle file as pem")
+		return err
 	}
-	if _, err = x509.ParseCertificate(block.Bytes); err != nil {
-		return nil, err
+	if !x509.NewCertPool().AppendCertsFromPEM(pemBlock) {
+		return fmt.Errorf("failed to find CA bundle in file %s", config.CABundleFile)
```
Comment (stlaz) [inline pkg/controller/configmapcainjector/starter/starter.go]: Please keep the original error as well, while there's probably very little that can go wrong while trying to parse PEM block, I'd still say there would be cases where having the original error could prove helpful.

File: pkg/generated/clientset/versioned/typed/serviceca/v1/fake/fake_serviceca.go [hunk 2]
```diff
@@ -123,7 +123,7 @@ func (c *FakeServiceCAs) DeleteCollection(options *v1.DeleteOptions, listOptions
 // Patch applies the patch and returns the patched serviceCA.
 func (c *FakeServiceCAs) Patch(name string, pt types.PatchType, data []byte, subresources ...string) (result *servicecav1.ServiceCA, err error) {
 	obj, err := c.Fake.
-		Invokes(testing.NewRootPatchSubresourceAction(servicecasResource, name, pt, data, subresources...), &servicecav1.ServiceCA{})
+		Invokes(testing.NewRootPatchSubresourceAction(servicecasResource, name, data, subresources...), &servicecav1.ServiceCA{})
```
Comment (stlaz) [inline pkg/generated/clientset/versioned/typed/serviceca/v1/fake/fake_serviceca.go]: Please put generated changes into a separate commit.

File: pkg/operator/sync_all_v4_00.go [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
 import (
+	"github.com/openshift/library-go/pkg/operator/events"
 	scsv1 "github.com/openshift/service-ca-operator/pkg/apis/serviceca/v1"
 )
 
 // sync_v4_00_to_latest takes care of synchronizing (not upgrading) the thing we're managing.
 // most of the time the sync method will be good for a large span of minor versions
-func sync_v4_00_to_latest(c serviceCertSignerOperator, operatorConfig *scsv1.ServiceCA) error {
-	err := syncSigningController_v4_00_to_latest(c, operatorConfig)
+func sync_v4_00_to_latest(c serviceCAOperator, eventRecorder events.Recorder, operatorConfig *scsv1.ServiceCA) error {
```
Comment (stlaz) [inline pkg/operator/sync_all_v4_00.go]: Why add the `eventRecorder` when you're passing the whole `serviceCAOperator` struct which has it included?

File: pkg/operator/sync_apiservice_v4_00.go [hunk 4]
```diff
@@ -10,50 +11,52 @@ import (
 	appsclientv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
 	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
 
+	"github.com/openshift/library-go/pkg/operator/events"
 	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
 	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
 	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
 	scsv1 "github.com/openshift/service-ca-operator/pkg/apis/serviceca/v1"
```
Comment (stlaz) [inline pkg/operator/sync_apiservice_v4_00.go]: nit: this line should probably be grouped with those below, too

File: pkg/operator/sync_apiservice_v4_00.go [hunk 5]
```diff
@@ -10,50 +11,52 @@ import (
...
+
+	"github.com/openshift/service-ca-operator/pkg/operator/operatorclient"
 	"github.com/openshift/service-ca-operator/pkg/operator/v4_00_assets"
-	"os"
 )
 
 // syncAPIServiceController_v4_00_to_latest takes care of synchronizing (not upgrading) the thing we're managing.
 // most of the time the sync method will be good for a large span of minor versions
-func syncAPIServiceController_v4_00_to_latest(c serviceCertSignerOperator, operatorConfig *scsv1.ServiceCA) error {
+func syncAPIServiceController_v4_00_to_latest(c serviceCAOperator, eventRecorder events.Recorder, operatorConfig *scsv1.ServiceCA) error {
```
Comment (stlaz) [inline pkg/operator/sync_apiservice_v4_00.go]: Extra event recorder being passed here.

File: pkg/operator/sync_configmapcabundle_v4_00.go [hunk 6]
```diff
@@ -10,47 +11,49 @@ import (
...
 	scsv1 "github.com/openshift/service-ca-operator/pkg/apis/serviceca/v1"
+	"github.com/openshift/service-ca-operator/pkg/operator/operatorclient"
 	"github.com/openshift/service-ca-operator/pkg/operator/v4_00_assets"
-	"os"
 )
 
 // syncConfigMapCABundleController_v4_00_to_latest takes care of synchronizing (not upgrading) the thing we're managing.
 // most of the time the sync method will be good for a large span of minor versions
-func syncConfigMapCABundleController_v4_00_to_latest(c serviceCertSignerOperator, operatorConfig *scsv1.ServiceCA) error {
+func syncConfigMapCABundleController_v4_00_to_latest(c serviceCAOperator, eventRecorder events.Recorder, operatorConfig *scsv1.ServiceCA) error {
```
Comment (stlaz) [inline pkg/operator/sync_configmapcabundle_v4_00.go]: Extra event recorder being passed here.

File: pkg/operator/sync_signer_v4_00.go [hunk 7]
```diff
@@ -13,50 +14,52 @@ import (
...
 	scsv1 "github.com/openshift/service-ca-operator/pkg/apis/serviceca/v1"
+	"github.com/openshift/service-ca-operator/pkg/operator/operatorclient"
 	"github.com/openshift/service-ca-operator/pkg/operator/v4_00_assets"
-	"os"
 )
 
 // syncSigningController_v4_00_to_latest takes care of synchronizing (not upgrading) the thing we're managing.
 // most of the time the sync method will be good for a large span of minor versions
-func syncSigningController_v4_00_to_latest(c serviceCertSignerOperator, operatorConfig *scsv1.ServiceCA) error {
+func syncSigningController_v4_00_to_latest(c serviceCAOperator, eventRecorder events.Recorder, operatorConfig *scsv1.ServiceCA) error {
```
Comment (stlaz) [inline pkg/operator/sync_signer_v4_00.go]: Extra event recorder being passed here.

File: pkg/operator/sync_apiservice_v4_00.go (hunk 5 above)
Comment (ericavonb) [reply pkg/operator/sync_apiservice_v4_00.go]: what do you mean extra? It's required to pass into the resourceapply functions.

File: manifests/0000_09_service-ca-operator_07_clusteroperator.yaml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+apiVersion: config.openshift.io/v1
+kind: ClusterOperator
+metadata:
+  name: service-ca
+spec: {}
```
[the remark this answers is not in the document]
Comment (ericavonb) [reply manifests/0000_09_service-ca-operator_07_clusteroperator.yaml]: removed in 16f36c01 (will issue another pr)

File: manifests/0000_09_service-ca-operator_05_deploy.yaml [hunk 9]
```diff
@@ -26,7 +26,7 @@ spec:
         - "--config=/var/run/configmaps/config/operator-config.yaml"
         - "-v=4"
         env:
-        - name: CONTROLLER_IMAGE
+        - name: OPERATOR_IMAGE
```
[the remark this answers is not in the document]
Comment (ericavonb) [reply manifests/0000_09_service-ca-operator_05_deploy.yaml]: changed back in 16f36c01

File: pkg/controller/configmapcainjector/starter/starter.go (hunk 1 above)
Comment (ericavonb) [reply pkg/controller/configmapcainjector/starter/starter.go]: changed back to original in 16f36c01

File: pkg/operator/sync_apiservice_v4_00.go (hunk 5 above)
Comment (stlaz) [reply pkg/operator/sync_apiservice_v4_00.go]: https://github.com/openshift/service-ca-operator/pull/16#discussion_r259325393

File: pkg/generated/clientset/versioned/typed/serviceca/v1/fake/fake_serviceca.go (hunk 2 above)
Comment (ericavonb) [reply pkg/generated/clientset/versioned/typed/serviceca/v1/fake/fake_serviceca.go]: now in 03446ad1

File: pkg/operator/sync_all_v4_00.go (hunk 3 above)
Comment (ericavonb) [reply pkg/operator/sync_all_v4_00.go]: 🤦‍♀️ removed

File: bindata/v4.0.0/service-ca-operator/operator-config.yaml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+apiVersion: operator.openshift.io/v1
+kind: ServiceCA
+metadata:
+  name: cluster
+spec:
+  managementState: Managed
```
[the remark this answers is not in the document]
Comment (stlaz) [reply bindata/v4.0.0/service-ca-operator/operator-config.yaml]: Never mind, I realized this won't be applied unless the config is missing, my bad.
